app/permissions/experiment.rb

```diff
@@ -88,12 +89,18 @@
     user.is_owner_of_project?(my_module.experiment.project)
   end
 
-  # module: assign/unassign sample, assign/unassign repository record
+  # module: assign/unassign repository record
   # NOTE: Use 'module_page? &&' before calling this permission!
   can :assign_repository_rows_to_module do |user, my_module|
     user.is_technician_or_higher_of_project?(my_module.experiment.project)
   end
 
+  # module: assign/unassign sample
+  # NOTE: Use 'module_page? &&' before calling this permission!
+  can :assign_sample_to_module do |user, my_module|
```

Comment:
Why the duplication of the same permission? Maybe we should rename this to manage_repository_rows or something ask the permission queen @mlorb :smile:

Reply:
I agree. Samples will be removed anyway. We can use `assign_repository_rows_to_module` permission also for samples. There's no need for duplication of code.

Reply:
we will remove all samples permissions once we remove samples